File: CarlaErik/UvA.SoftCon.Questionnaire/QL/AST/Model/Statements/IfStatement.cs
using System.Collections.Generic;
using UvA.SoftCon.Questionnaire.Common.AST.Model;
using UvA.SoftCon.Questionnaire.QL.AST.Model.Expressions;

namespace UvA.SoftCon.Questionnaire.QL.AST.Model.Statements
{
    public class IfStatement : Statement
    {
        public Expression If
        {
            get;
            private set;
        }

        public IEnumerable<Statement> Then
        {
            get;
            private set;
        }

        public IEnumerable<Statement> Else
        {
            get;
            private set;
        }

        internal IfStatement(Expression @if, IEnumerable<Statement> then, IEnumerable<Statement> @else, TextPosition position)
            : base(position)
        {
            If = @if;
            Then = then;
            Else = @else;
        }

        public override T Accept<T>(IQuestionFormVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }

        internal override void CollectQuestions(ICollection<Question> questions)
        {
            foreach (var statement in Then)
            {
                statement.CollectQuestions(questions);
            }
            foreach (var statement in Else)
            {
                statement.CollectQuestions(questions);
            }
        }
    }
}

File: Zisimopoulos_Valencia/Software Construction/QL/src/uva/sc/ql/statements/IfStatement.java
package uva.sc.ql.statements;

import java.util.List;

import uva.sc.ql.ast.IQLExpressionNodeVisitor;
import uva.sc.ql.ast.IQLStatementNodeVisitor;
import uva.sc.ql.expression.Expression;

public class IfStatement implements Statement {

    Expression expr;
    List<Question> questions;

    public IfStatement(Expression expr, List<Question> questions) {
	this.expr = expr;
	this.questions = questions;
    }

    public Expression getExpr() {
	return expr;
    }

    public List<Question> getQuestions() {
	return questions;
    }

    public String toString() {
	String result = "[IF] (";
	if (this.expr != null) {
	    result += this.expr.toString() + ") THEN {";
	}
	for (int i = 0; i < this.questions.size(); i++) {
	    result += questions.get(i).toString() + ",";
	}
	return result + "}";
    }

    public <T> T accept(IQLStatementNodeVisitor<T> visitor) {
	return visitor.visit(this);
    }

}
